### PR TITLE
add basic search view for notifications

### PR DESCRIPTION
**Pull request by richard-jones**
# Search and short notifications view

For: https://github.com/DOAJ/doajPM/issues/3282

Adds a short notification view in the navigation, and a full notifications search view

## Categorisation

This PR...
- [ ] has scripts to run
- [ ] has migrations to run
- [ ] adds new infrastructure
- [ ] changes the CI pipeline
- [ ] affects the public site
- [x] affects the editorial area
- [x] affects the publisher area (specifically, some publisher notifications change slightly)

## Basic PR Checklist

- [x] FeatureMap annotations have been added
- [x] Unit tests have been added/modified
- [x] Functional tests have been added/modified
- [x] Code has been run manually in development, and functional tests followed locally
- [x] No deprecated methods are used
- [x] No magic strings/numbers - all strings are in `constants` or `messages` files
- [x] Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
- [x] **N/A** If needed, migration has been created and tested locally
- [x] Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit#gid=753918108&range=A2
- [x] Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
    - [x] **N/A** Core model documentation: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
    - [x] Events and consumers documentation: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit

## Testing

To test this we would want to run through all the notifications and confirm that they are generated as expected, and appear in the user interface as expected.

Functional tests to generate specific notifications are listed in the documentation here: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit#gid=1277065021&range=O2

## Deployment

No deployment considerations

### CODE REVIEW

**Review by philipkcl — CHANGES_REQUESTED** [review]

File: portality/lib/edges.py [hunk 1]
```diff
@@ -7,10 +7,15 @@
 
 
 def make_url_query(**params):
-    query = make_query(**params)
+    query = make_query_json(**params)
     return parse.quote_plus(json.dumps(query))
```
[the remark this answers is not in the document]
Comment (richard-jones) [reply portality/lib/edges.py:11]: This is fixed on `master`, so the fix should bubble up during merge

**Review by philipkcl — COMMENTED** [review]
one comment for `url_for`

File: portality/events/consumers/application_publisher_accepted_notify.py [hunk 2]
```diff
@@ -64,8 +64,8 @@ def consume(cls, event):
         notification.long = svc.long_notification(cls.ID).format(
             application_title=application.bibjson().title,
             application_date=dates.human_date(application.date_applied),
-            publisher_dashboard_url=app.config.get('BASE_URL', "https://doaj.org") + url_for("publisher.journals"),
-            faq_url=app.config.get('BASE_URL', "https://doaj.org") + url_for("doaj.faq")
+            publisher_dashboard_url=app.config.get("BASE_URL") + url_for("publisher.journals"),
+            faq_url=url_for("doaj.faq", _external=True)
```
[the remark this answers is not in the document]
Comment (richard-jones) [reply portality/events/consumers/application_publisher_accepted_notify.py:68]: Yes, good spot, I'll fix that one too

[the remark this answers is not in the document]
Comment (richard-jones) [reply portality/events/consumers/application_publisher_accepted_notify.py]: Fixed

**Comment by richard-jones** [comment]
I have fixed the issues identified and also merged up from develop (via feature/editoria-workflow--stage2)

**Review by amdomanska — APPROVED** [review]
I couldn't generate notifications but the button looks slightly out of line, is that intentional?

image


Other than that looks fine to me.

**Review by Steven-Eardley — APPROVED** [review]
This is superb

**Pull request merged**